libs/tdigest/src/main/java/org/elasticsearch/tdigest/arrays/WrapperTDigestArrays.java

```diff
@@ -0,0 +1,276 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ * This project is based on a modification of https://github.com/tdunning/t-digest which is licensed under the Apache 2.0 License.
+ */
+
+package org.elasticsearch.tdigest.arrays;
+
+import org.apache.lucene.util.ArrayUtil;
+
+import java.util.Arrays;
+
+/**
+ * Temporal TDigestArrays with raw arrays.
+ *
+ * <p>
+ *     Delete after the right implementation for BigArrays is made.
+ * </p>
```

Comment:
I think we'd move this implementation to the tests rather than delete.

Reply:
Was thinking of that, but the BigArrays iml with MockBigArrays has the side effect of checking for leaks. So it feels better to me to just keep that implementation. But I'll consider it in the next PR

Reply:
Ah, yeah, we could indeed just do that.